ui/nuxeo-layout.js

```diff
@@ -125,7 +122,17 @@ import './nuxeo-error.js';
           valid = (el.validate ? el.validate() : el.checkValidity()) && valid;
         }
       }
-      return valid;
+      if (!valid) {
+        return false;
+      }
+      if (this.element && typeof this.element.validate === 'function') {
+        const elValidate = this.element.validate();
```

Comment:
Now that I think about this 🤔 , since we are in an async function, I think we could just return `this.element.validate()` right away. I don't think we need to wait for it to be evaluated, since people will have to await for the validate method anyway.
And if the result is not a promise, the value will be wrapped a promise.

Reply:
Indeed! :+1: